File: services/frontend/src/lib/api/applicants/types/applicant.ts
import type { ApplicantAddress, ApplicantContact, ApplicantTrainingsAddress } from ".";

export interface ApplicantResponse {
  maxCount: number;
  count: number;
  applicants: Applicant[];
}

export interface Applicant {
  id: string;
  firstname: string;
  lastname: string;

  balanceOutgoing: number | null;
  balancePayback: number | null;

  address: ApplicantAddress;
  trainingsAddress?: ApplicantTrainingsAddress;
  contact: ApplicantContact;
}

File: services/backend/src/resources/applicants/service/models/applicant.go
package models

import "github.com/google/uuid"

type ApplicantModel struct {
	ID        uuid.UUID
	Firstname string
	Lastname  string

	BalancePayback  *float64
	BalanceOutgoing *float64

	Address          *ApplicantAddressModel
	TrainingsAddress *ApplicantTrainingsAddressModel
	Contact          *ApplicantContactModel
}

type ApplicantInsertModel struct {
	Firstname string
	Lastname  string

	AddressID     uuid.UUID
	ContactID     uuid.UUID
	BankAccountID uuid.UUID
}

type ApplicantBankAccountModel struct {
	Iban          string
	Bic           string
	BankName      string
	AccountHolder string
}

type ApplicantAddressModel struct {
	Street      string
	HouseNumber string
	ZipCode     string
	City        string
	Country     string
}

type ApplicantTrainingsAddressModel struct {
	Street      string
	HouseNumber string
	ZipCode     string
	City        string
	Country     string
}

type ApplicantContactModel struct {
	Phone string
	Email string
}
